src/Command/CompileCommand.php

```diff
@@ -0,0 +1,135 @@
+     * @var \Cake\TwigView\View\TwigView
+     */
+    private $twigView;
+
+    /**
+     * @inheritdoc
+     */
+    protected function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser->addArgument('type', [
+            'required' => true,
+            'choices' => ['all', 'file', 'plugin'],
+            'help' => 'The type you want to compile.',
+        ]);
+
+        $parser->addArgument('target', [
+            'required' => false,
+            'help' => 'The file or plugin you want to compile.',
+        ]);
+
+        return $parser;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(Arguments $args, ConsoleIo $io): ?int
+    {
+        $type = $args->getArgumentAt(0);
+        $target = $args->getArgumentAt(1);
+
+        $this->twigView = new TwigView();
+        switch ($type) {
+            case 'all':
+                $this->compileAll($io);
+                break;
+
+            case 'file':
+                if ($target === null) {
+                    $io->error('File name is missing.');
+
+                    return static::CODE_ERROR;
+                }
+
+                $this->compileFile($io, $target);
+                break;
+
+            case 'plugin':
+                if ($target === null) {
+                    $io->error('Plugin name is missing.');
+
+                    return static::CODE_ERROR;
+                }
+
+                $this->compilePlugin($io, $target);
+                break;
+
+            default:
+                return static::CODE_ERROR;
+        }
+
+        return static::CODE_SUCCESS;
+    }
+
+    /**
+     * Compile all templates.
+     *
+     * @param \Cake\Console\ConsoleIo $io The logger
+     * @return void
+     */
+    private function compileAll(ConsoleIo $io): void
+    {
+        $io->info('Compiling all templates');
+
+        foreach (Scanner::all() as $section => $templates) {
+            $io->info("Compiling section {$section}");
+            foreach ($templates as $template) {
+                $this->compileFile($io, $template);
+            }
+        }
+    }
+
+    /**
+     * Compile all templates for a plugin.
+     *
+     * @param \Cake\Console\ConsoleIo $io The logger
+     * @param string $plugin The plugin name
+     * @return void
+     */
+    private function compilePlugin(ConsoleIo $io, string $plugin): void
+    {
+        $io->info("Compiling plugin {$plugin}");
+
+        foreach (Scanner::all() as $section => $templates) {
+            $io->info("Compiling template section {$section}");
+            foreach (Scanner::plugin($plugin) as $template) {
+                $this->compileFile($io, $template);
+            }
+        }
+    }
+
+    /**
+     * Compile a single template file.
+     *
+     * @param \Cake\Console\ConsoleIo $io The logger
+     * @param string $filename The file name
+     * @return void
+     */
+    private function compileFile(ConsoleIo $io, string $filename): void
```

Comment:
does it need to be private? What if one wants to extend and overwrite a small part? This would prevent it.

Reply:
I don't mind making it protected. I thought we would make it protected if we ever needed to extend a class. Seems awkward to make very single function protected.

Reply:
All functions in such a class :)